ngrok-java/src/main/java/com/ngrok/Http.java

```diff
@@ -0,0 +1,388 @@
+package com.ngrok;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * A set of data classes to support creation of HTTP endpoint listeners.
+ */
+public interface Http {
+    /**
+     * Represents the scheme for an HTTP listener.
+     */
+    enum Scheme {
+        HTTP("HTTP"),
+        HTTPS("HTTPS");
+
+        /**
+         * The name of the scheme.
+         */
+        public final String name;
+
+        Scheme(String name) {
+            this.name = name;
+        }
+    }
+
+    /**
+     * Represents an HTTP header.
+     */
+    class Header {
+        private final String name;
+        private final String value;
+
+        /**
+         * Constructs a new header with the specified name and value.
+         *
+         * @param name  the name of the header
+         * @param value the value of the header
+         */
+        public Header(String name, String value) {
+            this.name = Objects.requireNonNull(name);
+            this.value = Objects.requireNonNull(value);
+        }
+
+        /**
+         * Returns the name of the header.
+         *
+         * @return the name of the header
+         */
+        public String getName() {
+            return name;
+        }
+
+        /**
+         * Returns the value of the header.
+         *
+         * @return the value of the header
+         */
+        public String getValue() {
+            return value;
+        }
+    }
+
+    /**
+     * Represents basic authentication options for an HTTP listener.
+     */
+    class BasicAuth {
+        private final String username;
+        private final String password;
+
+        /**
+         * Constructs a new set of basic authentication options with the specified
+         * username and password.
+         *
+         * @param username the username
+         * @param password the password
+         */
+        public BasicAuth(String username, String password) {
+            this.username = Objects.requireNonNull(username);
+            this.password = Objects.requireNonNull(password);
+        }
+
+        /**
+         * Returns the username for basic auth.
+         *
+         * @return the username
+         */
+        public String getUsername() {
+            return username;
+        }
+
+        /**
+         * Returns the password for basic auth.
+         *
+         * @return the password
+         */
+        public String getPassword() {
+            return password;
+        }
+    }
+
+    /**
+     * Represents OAuth configuration for an HTTP listener.
+     */
+    class OAuth {
+
+        private final String provider;
+
+        private String clientId;
+
+        private String clientSecret;
+
+        private Optional<String> allowEmail = Optional.empty();
+
+        private Optional<String> allowDomain = Optional.empty();
+
+        private Optional<String> scope = Optional.empty();
```

Comment:
Could email/domain/scope be lists to match up with rust?

Reply:
Hm, didn't realize these are lists, will update 👍